parser_test.go

```diff
@@ -14,177 +14,190 @@ func TestExample(t *testing.T) {
+	bts, err := json.MarshalIndent(p.OpenAPI, "", "    ")
 	require.NoError(t, err)
 
 	expected := `
 {
-   "openapi":"3.0.0",
-   "info":{
-      "title":"LaunchDarkly REST API",
-      "description":"Build custom integrations with the LaunchDarkly REST API",
-      "contact":{
-         "name":"LaunchDarkly Technical Support Team",
-         "url":"https://support.launchdarkly.com",
-         "email":"support@launchdarkly.com"
-      },
-      "license":{
-         "name":"Apache 2.0",
-         "url":"https://www.apache.org/licenses/LICENSE-2.0"
-      },
-      "version":"2.0"
-   },
-   "servers":[
-      {
-         "url":"https://app.launchdarkly.com"
-      }
-   ],
-   "paths":{
-      "/api/v2/foo":{
-         "get":{
-            "responses":{
-               "200":{
-                  "description":"Successful foo response",
-                  "content":{
-                     "application/json":{
-                        "schema":{
-                           "$ref":"#/components/schemas/FooResponse"
-                        }
-                     }
-                  }
-               },
-               "401":{
-                  "description":"Invalid access token"
-               },
-               "403":{
-                  "description":"Forbidden"
-               },
-               "404":{
-                  "description":"Invalid resource identifier"
-               }
+    "components": {
+        "schemas": {
+            "Environment": {
+                "properties": {
+                    "name": {
+                        "type": "string"
+                    }
+                },
+                "type": "object"
             },
-            "summary":"Get all foos",
-            "description":" Get all foos"
-         },
-         "put":{
-            "responses":{
-               "200":{
-                  "description":"Successful foo response",
-                  "content":{
-                     "application/json":{
-                        "schema":{
-                           "$ref":"#/components/schemas/FooResponse"
-                        }
-                     }
-                  }
-               },
-               "401":{
-                  "description":"Invalid access token"
-               },
-               "403":{
-                  "description":"Forbidden"
-               },
-               "404":{
-                  "description":"Invalid resource identifier"
-               }
+            "FooResponse": {
+                "properties": {
+                    "environments": {
+                        "additionalProperties": {
+                            "properties": {
+                                "name": {
+                                    "type": "string"
+                                }
+                            },
+                            "type": "object"
+                        },
+                        "type": "object"
+                    },
```

Comment:
here's the generated portion. we can see that there's an unrelated bug; it regenerates the schema object instead of using a `$ref` to `Environment`